src/compiler/scala/tools/nsc/typechecker/Implicits.scala

```diff
@@ -473,12 +473,17 @@ trait Implicits {
         val syms = for (t <- tp; if t.typeSymbol.isTypeParameter) yield t.typeSymbol
         deriveTypeWithWildcards(syms.distinct)(tp)
       }
+      def sumComplexity(ts: List[Type]): Int = {
+        var s = 0
+        ts foreach { t => s += complexity(t) }
+        s
+      }
```

Comment:
You could make it more loopy:
```
@annotation.tailrec def g(sum: Int, xs: List[Type]): Int = xs match {
  case h :: t => g(sum + complexity(h), t)
  case _: Nil.type => sum 
}
```
Your `s` is captured, so it's off-stack. You could also consider a while loop and allocating an iterator, but that would not be as fancy.